public/components/model/model_details_panel.tsx

```diff
@@ -3,17 +3,242 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import React from 'react';
-import { EuiHorizontalRule, EuiPanel, EuiSpacer, EuiTitle } from '@elastic/eui';
+import React, { useState, useCallback, useEffect, useRef, useMemo } from 'react';
+import {
+  EuiButton,
+  EuiCallOut,
+  EuiDescribedFormGroup,
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiForm,
+  EuiFormRow,
+  EuiHorizontalRule,
+  EuiLink,
+  EuiPanel,
+  EuiSpacer,
+  EuiText,
+  EuiTextArea,
+  EuiTitle,
+  htmlIdGenerator,
+} from '@elastic/eui';
+import { useController, useForm } from 'react-hook-form';
+import { generatePath, useHistory } from 'react-router-dom';
+
+import { useOpenSearchDashboards } from '../../../../../src/plugins/opensearch_dashboards_react/public';
+import { mountReactNode } from '../../../../../src/core/public/utils';
+import { routerPaths } from '../../../common';
+
+import { BottomFormActionBar } from './bottom_form_action_bar';
+
+const NAME_MAX_LENGTH = 80;
+
+const DESCRIPTION_MAX_LENGTH = 200;
+
+interface ModelDetailsProps {
+  id: string;
+  name?: string;
+  description?: string;
+  onDetailsUpdate?: (formData: { name: string; description?: string }) => void;
+}
+
+export const ModelDetailsPanel = ({
+  id,
+  name,
+  description,
+  onDetailsUpdate,
+}: ModelDetailsProps) => {
+  const formIdRef = useRef(htmlIdGenerator()());
+  const history = useHistory();
+  const { control, resetField, formState, handleSubmit } = useForm({
+    mode: 'onChange',
+    defaultValues: {
+      name: '',
+      description: '',
+    },
+  });
+  const [isEditMode, setIsEditMode] = useState(false);
+  const {
+    services: { notifications },
+  } = useOpenSearchDashboards();
+
+  const nameController = useController({
+    control,
+    name: 'name',
+    rules: {
+      required: { value: true, message: 'Name can not be empty' },
```

Comment:
Since the unique check logic and API is not finalized yet, this validation logic may move to backend API phase. I think we can add this check after confirmed.